org.testeditor.web.dropwizard.xtext/src/test/java/org/testeditor/web/xtext/index/BuildCycleManagerTest.xtend

```diff
@@ -0,0 +1,197 @@
+	@Mock ChangeDetector mockChangeDetector
+	@Mock ChangedResources mockChangedResources
+	@Mock XtextResourceSet mockResourceSet
+	@Mock ValidationMarkerUpdater mockValidationMarkerUpdater
+	@Mock IncrementalBuilder mockIncrementalBuilder
+	@Mock extension IResourceServiceProvider.Registry mockResourceServiceProviderRegistry
+	@Mock ChunkedResourceDescriptionsProvider indexProvider
+	@Mock ChunkedResourceDescriptions index
+	@InjectMocks extension BuildCycleManager unitUnderTest
+
+	val expectedModifiedResources = #[URI.createFileURI('/path/to/modified/resource')]
+	val expectedDeletedResource = #[URI.createFileURI('/path/to/deleted/resource')]
+	val initialIndexState = new IndexState
+
+	var BuildRequest sampleBuildRequest
+
+	@Before
+	def void setupChangeDetectorMock() {
+		MockitoAnnotations.initMocks(this)
+
+		sampleBuildRequest = new BuildRequest => [
+			baseDir = URI.createFileURI(config.localRepoFileRoot)
+			resourceSet = mockResourceSet
+			afterValidate = mockValidationMarkerUpdater
+			deletedFiles = expectedDeletedResource
+			dirtyFiles = expectedModifiedResources
+			state = initialIndexState
+		]
+
+		when(mockSearchPathProvider.additionalSearchPaths(config.localRepoFileRoot)).thenReturn(#['/base/dir/some.jar'])
+		when(mockChangeDetector.detectChanges(eq(mockResourceSet), any)).thenReturn(mockChangedResources)
+		when(mockChangedResources.modifiedResources).thenReturn(expectedModifiedResources)
+		when(mockChangedResources.deletedResources).thenReturn(expectedDeletedResource)
+
+		val builderResult = mock(IncrementalBuilder.Result)
+		val indexState = getMockedIndexState(#['Test'])
+		when(builderResult.indexState).thenReturn(indexState)
+		when(mockIncrementalBuilder.build(eq(sampleBuildRequest), any)).thenAnswer [
+			<BuildRequest>getArgument(0).afterValidate.afterValidate(null, null)
+			builderResult
+		]
+
+		when(indexProvider.getIndex(mockResourceSet)).thenReturn(index)
+	}
+
+	private def getMockedIndexState(Iterable<String> eObjectNames) {
+		val indexState = mock(IndexState)
+		val resourceDescriptionsData = mock(ResourceDescriptionsData)
+		when(resourceDescriptionsData.exportedObjects).thenReturn(
+			eObjectNames.map [ eObjectName |
+			val desc = new SerializableEObjectDescription()
+			desc.qualifiedName = QualifiedName.create(eObjectName)
+			return desc
+		])
+		when(indexState.resourceDescriptions).thenReturn(resourceDescriptionsData)
+		return indexState
+	}
+
+	@Test
+	def void detectChangesReturnsModifiedFiles() {
+		// given
+		val initialBuildRequest = new BuildRequest
+
+		// when
+		val actualBuildRequest = initialBuildRequest.addChanges
+
+		// then
+		assertThat(actualBuildRequest.deletedFiles).containsOnly(expectedDeletedResource)
+		assertThat(actualBuildRequest.dirtyFiles).containsOnly(expectedModifiedResources)
+		verify(mockSearchPathProvider).additionalSearchPaths(config.localRepoFileRoot)
+	}
+
+	@Test
+	def void createBuildRequestSetsRequiredFields() {
```

Comment:
testing simple object construction seems a bit too much here, doesn't it?

Reply:
Yeah. As stated initially, this whole test is somewhat questionable, but it was a useful tool during creation of the corresponding new class (in terms of TDD). I would argue that this test still has value in terms of documenting the expected post-conditions of the method under test. BuildRequest is a mutable object, and it needs to be put into a specific state for subsequent steps to work as intended. So, in terms of evolvability, it might still be useful to keep this test; without it, problems due to a violation of the underlying contract may only bubble up much later, potentially only at runtime (e.g. validation markers are not updated anymore).

… that said, I'm just as happy to delete the test, if you prefer that. :)